crates/bevy_reflect/src/list.rs

```diff
@@ -7,17 +7,42 @@ use crate::{
     ReflectRef, TypeInfo, Typed,
 };
 
-/// An ordered, mutable list of [Reflect] items. This corresponds to types like [`std::vec::Vec`].
+/// A trait used to power [list-like] operations via [reflection].
 ///
-/// This is a sub-trait of [`Array`], however as it implements [insertion](List::insert) and [removal](List::remove),
-/// it's internal size may change.
+/// This corresponds to types, like [`Vec`], which contain an ordered sequence
+/// of elements that implement [`Reflect`].
 ///
-/// This trait expects index 0 to contain the _front_ element.
-/// The _back_ element must refer to the element with the largest index.
-/// These two rules above should be upheld by manual implementors.
+/// This is a subtrait of [`Array`].
+/// However, it contains methods which allow its internal size to change,
+/// such as [`insert`](List::insert) and [`remove`](List::remove).
 ///
 /// [`push`](List::push) and [`pop`](List::pop) have default implementations,
-/// however it may be faster to implement them manually.
+/// however it will generally be more performant to implement them manually.
```

Comment:
```suggestion
/// however it will generally be more performant to implement them manually
/// as the default approach uses a very naive approach to find the correct position.
```